js/content/community.js

```diff
@@ -653,147 +657,141 @@ let ProfileHomePageClass = (function(){
         if (node) {
-            node.style.backgroundImage = "url(" + bg + ")";
-            return;
+            node.style.backgroundImage = `url("${bg}")`;
+        } else {
+            profilePage.classList.add("has_profile_background");
+            node = profilePage.querySelector(".profile_content");
+            node.classList.add("has_profile_background");
+            HTML.afterBegin(node,
+                `<div class="profile_background_holder_content">
+                    <div class="profile_background_overlay_content"></div>
+                    <div class="profile_background_image_content" style="background-image: url('${bg}');"></div>
+                </div>`);
         }
-
-        document.querySelector(".no_header").classList.add("has_profile_background");
-        node = document.querySelector(".profile_content");
-        node.classList.add("has_profile_background");
-        HTML.afterBegin(node, '<div class="profile_background_holder_content"><div class="profile_background_overlay_content"></div><div class="profile_background_image_content " style="background-image: url(' + bg + ');"></div></div></div>');
     };
 
     ProfileHomePageClass.prototype.addProfileStoreLinks = function() {
-        let nodes = document.querySelectorAll(".game_name .whiteLink");
-        for (let i=0, len=nodes.length; i<len; i++) {
-            let node = nodes[i];
+        for (let node of document.querySelectorAll(".game_name .whiteLink")) {
             let href = node.href.replace("//steamcommunity.com", "//store.steampowered.com");
-            HTML.afterEnd(node, "<br><a class='whiteLink' style='font-size: 10px;' href=" + href + ">" + Localization.str.visit_store + "</a>");
+            HTML.afterEnd(node, `<br><a class="whiteLink" style="font-size: 10px;" href="${href}">${Localization.str.visit_store}</a>`);
         }
     };
 
     ProfileHomePageClass.prototype.addSteamRepApi = function() {
         if (!SyncedStorage.get("showsteamrepapi")) { return; }
 
-        ProfileData.promise().then(data => {
-            if (!data.steamrep || data.steamrep.length === 0) { return; }
+        let steamrep = ProfileData.getSteamRep();
+        if (!steamrep || steamrep.length === 0) { return; }
 
-            let steamId = SteamId.getSteamId();
-            if (!steamId) { return; }
-
-            // Build reputation images regexp
-            let repImgs = {
-                "banned": /scammer|banned/gi,
-                "valve": /valve admin/gi,
-                "caution": /caution/gi,
-                "okay": /admin|middleman/gi,
-                "donate": /donator/gi
-            };
-
-            let html = "";
+        // Build reputation images regexp
+        let repImgs = {
+            "banned": /scammer|banned/i,
+            "valve": /valve admin/i,
+            "caution": /caution/i,
+            "okay": /admin|middleman/i,
+            "donate": /donator/i
```

Comment:
Removed the global flag, since we're only testing for an occurrence of the keywords